From 4f3ff8553c1011c23b59e8c32b0bad0c121d4954 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Tue, 12 Sep 2023 09:51:48 -0500
Subject: [PATCH 100/100] Address flake8 E721 violations (#32)

These violations don't show up in our CI right now because we've pinned
flake8<6, but when that pin is dropped, we'll have enforcement for this
rule.

Upstream-Status: Backport [https://github.com/colcon/colcon-metadata/commit/4f3ff8553c1011c23b59e8c32b0bad0c121d4954]

---
 colcon_metadata/metadata/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/colcon_metadata/metadata/__init__.py b/colcon_metadata/metadata/__init__.py
index de9ec63..1794afb 100644
--- a/colcon_metadata/metadata/__init__.py
+++ b/colcon_metadata/metadata/__init__.py
@@ -106,7 +106,7 @@ def merge_metadata_value(dict_, key, value, meta_source):
         old_value |= value
         return
 
-    if old_value is not None and type(old_value) != type(value):
+    if old_value is not None and type(old_value) is not type(value):
         logger.warning(
             "metadata from '{meta_source}' overwrites '{key}' with different "
             "type from value '{old_value}' to '{value}'".format_map(locals()))
-- 
2.47.3

